saw-central/src/SAWCentral/Crucible/MIR/Builtins.hs

```diff
@@ -2275,10 +2275,10 @@ setupResultTerm sc cc mty0 tpr0 val0 =
                          [ "ArrayShape with non-TyArray type:"
                          , Text.pack $ show $ PP.pretty mty
                          ]
-              shpTypeTerm <- shapeToTerm sc shp
+              eltTypeTerm <- shapeToTerm sc eltShp
               typedElts <- accessMirAggregateArray sym eltSz eltShp len val $
                   \_off val' -> go eltTy (shapeType eltShp) val'
-              scVectorReduced sc shpTypeTerm typedElts
+              scVectorReduced sc eltTypeTerm typedElts
```

Comment:
Nice find! I am surprised that `mir_extract` ever worked at all due to this oversight.